File: data/benchmark0/cpp/binary_search.cpp
#include <cstdint>

bool binary_search(int* a, int target, int n) {
    int lo = 0, hi = n - 1;
    while (lo <= hi) {
        int mid = lo + (hi - lo) / 2;
        if (a[mid] == target) return true;
        if (a[mid] < target) lo = mid + 1;
        else hi = mid - 1;
    }
    return false;
}

File: data/benchmark0/prepost/binary_search.rs
use vstd::prelude::*;

verus! {

pub fn binary_search(a: &[i32], target: i32) -> (result: bool)
    requires
        forall|i: int, j: int| 0 <= i < j < a@.len() ==> a@[i] <= a@[j],
    ensures
        result == exists|i: int| 0 <= i < a@.len() && a@[i] == target,
{
    assume(false);
}

} // verus!
